### PR TITLE
📊 growth: Update historical GDP to use new metadata

### PR DESCRIPTION
**Pull request by paarriagadap**
This is a continuation of the work described [here](https://github.com/owid/owid-issues/issues/379), but adding the new metadata structure.

### CODE REVIEW

**Review by veronikasamborska1994 — CHANGES_REQUESTED** [review]
hey @paarriagadap! :) I've taken a look at your code now! Most of it LGTM except this script `etl/steps/data/garden/growth/2024-01-04/gdp_historical.py` which I found to be quite confusing both in terms of code and your associated metadata description of the processing steps.

If you have some time I think it'd be worth spending a bit of time on improving your documentation of various steps in the code itself as well as the `description_processing`  field in the metadata as for someone with no background on this topic it was quite hard to follow.

File: etl/steps/data/garden/growth/2024-01-04/gdp_historical.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    # Process data.
+    tb = process_and_combine_datasets(tb_wdi, tb_mpd, tb_md)
+
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
```
[the remark this answers is not in the document]
Comment (paarriagadap) [reply etl/steps/data/garden/growth/2024-01-04/gdp_historical.py:57]: Yes, I am combining variables with different units, but I applied the growth of one of them retroactively to the other variable to keep consistency and to avoid mixing different things. The ETL (because of the metadata) still thinks I am proceeding wrong, but it's not the case. I hope the changes are clearer now.

File: etl/steps/data/garden/growth/2024-01-04/gdp_historical.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    # FOR MADDISON DATABASE
+    # Keep data until YEAR_MPD_MD
+    tb_md = tb_md[tb_md["year"] <= YEAR_MPD_MD].reset_index(drop=True)
+
+    # Drop population, as it's not needed
+    tb_md = tb_md.drop(columns=["population"])
+
+    #############################
+
+    # Merge both MPD and WDI world estimation
```
[the remark this answers is not in the document]
Comment (paarriagadap) [reply etl/steps/data/garden/growth/2024-01-04/gdp_historical.py]: I hope the code is clearer now. Yes, this is a way to stitch the three datasets that have different units.

Let me know if you still don't see I am aplpying the growth of the other datasets in the 1990 and 1820 stitches. You can see in these charts that the curves are parallel when plotting log axes https://github.com/owid/owid-issues/issues/379#issuecomment-1366749068

[the remark this answers is not in the document]
Comment (paarriagadap) [reply etl/steps/data/garden/growth/2024-01-04/gdp_historical.py]: I also tried to make description_processing clearer, hopefully that's the case.

File: etl/steps/data/garden/growth/2024-01-04/gdp_historical.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    Returns
+    -------
+    tb : Table
+        Table with the adjusted variables.
+    """
+    for var in var_list:
+        # Get value from the reference variable in the reference year
+        reference_value = tb.loc[tb["year"] == reference_year, f"{var}{reference_var_suffix}"].iloc[0]
+
+        # The scalar is the previous value divided by the reference variable. This is the growth that will be applied retroactively to the variable to be adjusted.
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/growth/2024-01-04/gdp_historical.py:171]: I think it's the other way around -  reference variable divided by the previous value but I now understand what this function does a lot better. thanks for making these changes!

File: etl/steps/data/garden/growth/2024-01-04/gdp_historical.meta.yml [hunk 4]
```diff
@@ -0,0 +1,128 @@
+# NOTE: To learn more about the fields, hover over their names.
+definitions:
+  common:
+    presentation:
+      topic_tags:
+        - Economic Growth
+    processing_level: major
+
+    description_processing: |-
+      The data presented here from 1990 onwards is from the World Bank. Data before 1990 is constructed using the 1990 World Bank observation and applying the growth rates implied by the Maddison Project Database. Similarly, data before 1820 is built using the adjusted 1820 estimate calculated by the previous method and applying the growth rates implied by the Maddison Database.
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/growth/2024-01-04/gdp_historical.meta.yml:10]: I think this could still be improved a bit - e.g., explain why you need to do this processing - ie. "Historical data is expressed in different units to World Bank, so to make it all consistent we can use the growth rates in our historical data and extrapolate back in time.. " or something along those lines. This is just to make it easy for everyone (and not just economists) to immediately understand the processing steps you've implemented but otherwise it all makes sense now. thanks for adding the comments

**Review by veronikasamborska1994 — APPROVED** [review]
LGTM

File: etl/steps/data/garden/growth/2024-01-04/gdp_historical.py (hunk 3 above)
Comment (paarriagadap) [reply etl/steps/data/garden/growth/2024-01-04/gdp_historical.py:171]: Ah, it's only that way because I am going backwards. But for clarity, I have inverted and then multiplied by the value to adjust, which is the same thing
